### PR TITLE
Improved: code to create a user search preference entry after creating search preference (#2e5ww56).

### PR DESCRIPTION
**Pull request by meet-aniket**
### Related Issues
 

 Closes #

 ### Short Description and Why It's Useful
 
:- Worked on improving the code to create a user search preference entry after creating search preference (#2e5ww56).

 ### Screenshots of Visual Changes before/after (If There Are Any)
 

 **IMPORTANT NOTICE** - Remember to add changelog entry

 ### Contribution and Currently Important Rules Acceptance
 

 - [ ] I read and followed [contribution rules](https://github.com/hotwax/threshold-management#contribution-guideline)

### CODE REVIEW

File: src/components/SaveThresholdModal.vue [hunk 1]
```diff
@@ -127,6 +127,13 @@ export default defineComponent({
 
         if (resp.status == 200 && resp?.data?.searchPrefId) {
           const searchPreferenceId = resp.data.searchPrefId;
+          const params = {
+            "searchPrefId": searchPreferenceId,
+            // TODO: Need to add userSearchPrefTypeId here to associate search preference with the user.
+            "userSearchPrefTypeId": ""
```
Comment (adityasharma7) [inline src/components/SaveThresholdModal.vue:133]:
```suggestion
            "userSearchPrefTypeId": "THRESHOLD_INV_QUERY"
```

File: src/components/SaveThresholdModal.vue [hunk 2]
```diff
@@ -127,6 +127,13 @@ export default defineComponent({
 
         if (resp.status == 200 && resp?.data?.searchPrefId) {
           const searchPreferenceId = resp.data.searchPrefId;
+          const params = {
+            "searchPrefId": searchPreferenceId,
+            // TODO: Need to add userSearchPrefTypeId here to associate search preference with the user.
```
Comment (adityasharma7) [inline src/components/SaveThresholdModal.vue:132]:
```suggestion
```

Comment (adityasharma7) [inline src/components/SaveThresholdModal.vue:132]:
```suggestion
```

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

**Review by ymaheshwari1 — APPROVED** [review]

File: src/components/SaveThresholdModal.vue [hunk 3]
```diff
@@ -127,6 +127,12 @@ export default defineComponent({
 
         if (resp.status == 200 && resp?.data?.searchPrefId) {
           const searchPreferenceId = resp.data.searchPrefId;
+          const params = {
+            "searchPrefId": searchPreferenceId,
+            "userSearchPrefTypeId": "THRESHOLD_INV_QUERY"
+          }
+          // Handle the case whether we will schedule service if searchPrefId is not associated with user. 
+          await ProductService.associateSearchPrefToUser(params);
```
Comment (adityasharma7) [inline src/components/SaveThresholdModal.vue:135]: I think we should handle error

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/components/SaveThresholdModal.vue (hunk 3 above)
Comment (shashwatbangar) [reply src/components/SaveThresholdModal.vue:135]: Done sir.

**Pull request merged**